team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java

```diff
@@ -123,38 +134,23 @@ private int playthrough(TreeNode current) {
 	}
 
 	private TreeNode getMaxLeaf(TreeNode root) {
-		ArrayList<TreeNode> leaves = getLeaves(root);
-
-		if (leaves.isEmpty())
-			return root;
-
-		int leafIndex = 0;
-		double maxUCB = leaves.get(0).getUCB();
-		for (int i = 1; i < leaves.size(); i++) {
-			double ucb = leaves.get(i).getUCB();
-			if (ucb > maxUCB) {
-				leafIndex = i;
-				maxUCB = ucb;
-			}
-		}
+		TreeNode node = root;
 
-		return leaves.get(leafIndex);
-	}
+		while (node.children.size() != 0) {
+			double maxUCB = Double.MIN_VALUE;
 
-	private ArrayList<TreeNode> getLeaves(TreeNode root) {
-		ArrayList<TreeNode> leaves = new ArrayList<>();
-
-		for (int i = 0; i < root.children.size(); i++) {
-			TreeNode current = root.children.get(i);
-			// Check if the child has children or is a leaf
-			if (current.children.isEmpty())
-				leaves.add(current);
-			else
-				// Recursively add current's leaves to the list of leaves
-				leaves.addAll(getLeaves(current));
-		}
+			TreeNode nodetoexpand = null;
 
-		return leaves;
```

Comment:
- Old getMaxLeaf was causing agent to commit suicide
- Recursion seems to be quite a bit slower (likely due to the increase in branching??)
- New method threw me a null exception VERY late in the game, there were still moves to be made, however, I had technically lost so I'm not sure if this is a real problem 
- Posted my spreadsheet in the discord